packages/api/src/api/extras/events.ts

```diff
@@ -37,3 +38,71 @@ export function isStateEvent(
 export function isRoomEvent(event: RoomEvent | StateEvent): event is RoomEvent {
   return !('state_key' in event);
 }
+
+/**
+ * Check if the given value is a valid {@link RoomEvent}.
+ *
+ * @param event - The value to check
+ * @returns true if value is a valid room event, else false.
+ */
+// Allow any here, so that the validation works for every event
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isValidRoomEvent(event: unknown): event is RoomEvent<any> {
+  return roomEventSchema.validate(event).error === undefined;
+}
+
+/**
+ * Check if the given value is a valid {@link StateEvent}.
+ *
+ * @param event - The value to check
+ * @returns true if value is a valid state event, else false.
+ */
+// Allow any here, so that the validation works for every event
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isValidStateEvent(event: unknown): event is StateEvent<any> {
+  return stateEventSchema.validate(event).error === undefined;
+}
+
+/**
+ * Check if the given value is a valid {@link ToDeviceMessageEvent}.
+ *
+ * @param event - The value to check
+ * @returns true if value is a valid to device message, else false.
+ */
+export function isValidToDeviceMessageEvent(
+  event: unknown,
+  // Allow any here, so that the validation works for every event
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): event is ToDeviceMessageEvent<any> {
+  return toDeviceMessageSchema.validate(event).error === undefined;
+}
+
+/**
+ * Base properties to validate for all events.
+ */
+const eventSchemaProps = {
+  type: Joi.string().required(),
+  // Do roughly check against the format
+  // https://spec.matrix.org/v1.13/appendices/#common-identifier-format
+  sender: Joi.string().pattern(new RegExp('^@.*:.*$')).required(),
```

Comment:
Similar as the room_id comment: Wouldn't something like `^@[^\s:]*:\S*$` with the `\S` and `\s` be more accurate here?

Reply:
Done